testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -3886,3 +3892,778 @@ fn antientropy_integration_test() {
+                    &atlasdb.conn,
+                    "SELECT content_hash FROM attachment_instances WHERE index_block_hash = ?1 AND attachment_index = ?2",
+                    &[ibh as &dyn ToSql, &u64_to_sql(*index).unwrap() as &dyn ToSql],
+                    "content_hash")
+                .unwrap();
+                if hashes.len() > 0 {
+                    assert_eq!(hashes.len(), 1);
+                    attachment_hashes.insert((ibh.clone(), *index), hashes.pop());
+                }
+            }
+        }
+    }
+    eprintln!("attachment_indexes = {:?}", &attachment_indexes);
+
+    for (ibh, attachments) in attachment_indexes.iter() {
+        let l = attachments.len();
+        for i in 0..(l / MAX_ATTACHMENT_INV_PAGES_PER_REQUEST + 1) {
+            if i * MAX_ATTACHMENT_INV_PAGES_PER_REQUEST >= l {
+                break;
+            }
+
+            let attachments_batch = attachments[i * MAX_ATTACHMENT_INV_PAGES_PER_REQUEST
+                ..cmp::min((i + 1) * MAX_ATTACHMENT_INV_PAGES_PER_REQUEST, l)]
+                .to_vec();
+            let path = format!(
+                "{}/v2/attachments/inv?index_block_hash={}&pages_indexes={}",
+                &http_origin,
+                ibh,
+                attachments_batch
+                    .iter()
+                    .map(|a| format!("{}", &a))
+                    .collect::<Vec<String>>()
+                    .join(",")
+            );
+
+            let attempts = 100;
+            let ts_begin = get_epoch_time_ms();
+            for _ in 0..attempts {
+                let res = client.get(&path).send().unwrap();
+
+                if res.status().is_success() {
+                    let attachment_inv_response: GetAttachmentsInvResponse = res.json().unwrap();
+                    eprintln!(
+                        "attachment inv response for {}: {:?}",
+                        &path, &attachment_inv_response
+                    );
+                } else {
+                    eprintln!("Bad response for `{}`: `{:?}`", &path, res.text().unwrap());
+                    panic!();
+                }
+            }
+            let ts_end = get_epoch_time_ms();
+            let total_time = ts_end.saturating_sub(ts_begin);
+            eprintln!("Requested {} {} times in {}ms", &path, attempts, total_time);
+
+            // requests should take no more than 20ms
+            assert!(
+                total_time < attempts * 40,
+                format!(
+                    "Atlas inventory request is too slow: {} >= {} * 40",
+                    total_time, attempts
+                )
+            );
+        }
+
+        for i in 0..l {
+            if attachments[i] == 0 {
+                continue;
+            }
+            let page_index = attachments[i] % 64; // AttachmentInstance::ATTACHMENTS_INV_PAGE_SIZE;
+            let content_hash = attachment_hashes
+                .get(&(*ibh, page_index))
```

Comment:
Looking at the insertions, I think it should be:
```suggestion
                .get(&(*ibh, attachments[i]))
```
Will test this patch locally.

Reply:
Ah, I was wondering why this wasn't working reliably.  I'll try it too -- I want to understand the system better.

Reply:
Confirming that this test is green on my machine, with this patch.

Reply:
Fixed here as well; let's see if the CI takes it.